cmd/sample-service/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+
+	"github.com/gorilla/mux"
+	"github.com/rturner3/spire-mysql-demo/pkg/common"
+	"github.com/rturner3/spire-mysql-demo/pkg/store"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	usersAPIPath = "/api/v1/users"
+
+	// SPIRE Agent socket path
+	socketPath = "unix:///run/spire/sockets/agent.sock"
+
+	mysqlUser   = "spire-mysql-client"
+	mysalDBName = "spiredemo"
+)
+
+type handler struct {
+	db      *sql.DB
+	dbStore *store.Store
+}
+
+func (h *handler) list(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	users, err := h.dbStore.ListUsers(r.Context())
+	if err != nil {
+		writeErr(w, err)
+		return
+	}
+
+	data, err := json.Marshal(users)
+	if err != nil {
+		writeErr(w, err)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(data)
+}
+
+func (h *handler) create(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		writeErr(w, err)
+		return
+	}
+
+	var user store.User
+	err = json.Unmarshal(body, &user)
+	if err != nil {
+		writeErr(w, err)
+		return
+	}
+
+	err = h.dbStore.CreateUser(r.Context(), user)
+	if err != nil {
+		writeErr(w, err)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+	w.Write([]byte(`{"message": "user created"}`))
+}
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Wait for an os.Interrupt signal
+	go waitForCtrlC(cancel)
+
+	// Creates a new Workload API client, connecting to provided socket path
+	// Environment variable `SPIFFE_ENDPOINT_SOCKET` is used as default
+	client, err := workloadapi.New(ctx, workloadapi.WithAddr(socketPath))
+	if err != nil {
+		log.Fatalf("Unable to create workload API client: %v", err)
+	}
+	defer client.Close()
+
+	x509Context, err := client.FetchX509Context(ctx)
+	if err != nil {
+		log.Fatalf("Unable to fetch x509Context %v", err)
+	}
+
+	db, err := common.NewMySQLDBWithSPIRETLSConfig(x509Context, mysqlUser, mysalDBName, "")
+	if err != nil {
+		log.Fatalf("Failed to create MySQL Client: %v", err)
+	}
+
+	h := &handler{
+		db:      db,
+		dbStore: store.New(db),
+	}
+
+	// Start X.509 watcher
+	go startWatcher(ctx, client, h)
+
+	log.Printf("Starting API handlers")
+	// Add API handlers
+	r := mux.NewRouter()
```

Comment:
Can we just use `net/http` for this? I don't see us doing anything too complicated with setting up the HTTP handler routes. I think it's better to keep a minimal dependency set for this sample to make it easier for people to understand

Reply:
I mean, this is pretty trivial code with `net/http`:
```go
http.HandleFunc(usersAPIPath, func(w http.ResponseWriter, req *http.Request) {
    switch req.Method {
    case http.MethodGet:
        h.list(w, req)
    case http.MethodPost:
        h.create(w, req)
    }
}
```

I don't think this specific dependency adds much cognitive load since it's fairly straightforward to comprehend without reading docs, so I'm not going to block on this, but my general feeling is that example code should use as few dependencies as necessary to keep it easier to read. Sometimes people who are less familiar with the Go language will get the idea that "this is how you're supposed to set up an HTTP server in Go", when in general the standard library is probably the more recommended path.